docs/src/main/asciidoc/security-openid-connect-multitenancy-concept.adoc

```diff
@@ -0,0 +1,317 @@
+:summary: This overview explains OpenID Connect multi-tenancy for Quarkus applications
+
+The goal of OIDC multi-tenancy is developing an application which can provide functionality to users from different organisations. Each enabled organisation could store its registered users and their credentials in its own `realm`, and use OIDC to supply security-related user information with the application. Each call to the application must therefore contain at least two kinds of information:
+
+* A `tenant identifier` which uniquely identifies the realm where the end user was authenticated / logged in. While you can technically use any means of sending the `tenant identifier` from the user's browser or other client-side application, it is convenient to insert it as a string within the context path of the service.
+* An `idToken` holding user information in the form of a JSON Web Token which contains a set of properties (called `claims` in OAuth2/OIDC vocabulary) for the authenticated user. Refer to https://jwt.io[JWT IO] for examples and  https://www.rfc-editor.org/rfc/rfc7519[RFC 7519] for the JWT specification.
+
+
+== The 1-minute overview of a multi-tenant application
+
+Multi-tenant applications can enable development teams to quickly provide and maintain functionality to many organisations without compromising security concerns. However, understanding and translating concepts and best practises from the OIDC world can be a daunting task, so this overview will illustrate the workings of a typical multi-tenant application to highlight the OIDC and security-related parts.
+
+.Two clients calling a multi-tenant application
+[caption="Figure 1: "]
+image::oidc-multitenancy-concept-1.svg[Multitenant Application]
+
+The image above shows 2 users from different organisations call a calendar service to find their respective organisations' festivities. For the sake of simplicity, we assume that the application serves RESTful data to a browser-based Single-Page Application (SPA).
+
+. Each user has logged in using their organisation's own single-sign ("SSO"), which is frequently called OIDC Authentication Provider (called `"OpenID Provider"`, `"Identity Provider"` or `"IdP"` in OIDC literature). Upon login, the SSO has produced an `IdToken` JWT with user data and delivered that to the user's web browser.
+. The two user will call the calendar service with _different resource paths_ within the service URL to reach the calendar service. The application uses the `tenant identifier` part of the URI to identify the caller's organisation. Should an authenticated user from organisation 1 accidentally use a resource path allocated to organisation 2, the call will yield an unauthenticated HTTP response.
+. The `IdToken` data retreived from the own organisation's SSO is normally sent to the multi-tenant application with each call as an HTTP header, called `"authorization:"`. In this respect, an IdToken works as a session cookie which contains user information.
```

Comment:
I'm not sure about `IdToken` references. SPA will get both ID and access tokens with the authorization code flow, with the traditional approach being is that it will itself use ID token to interact with the user but use the access token to access Quarkus to fetch something for the user.
SPA can surely post an ID token to Quarkus too, but unless a dedicated call like `useThisIDToken` is made to the application, as far as SPA passing ID token to Quarkus is concerned, it travels as a Bearer token - i.e Quarkus will treat it as the access token.
Note we have OIDC bearer and code flow guides - and I feel we should leave the aspects of dealing with the actual tokens to those guides (improve it as needed there) - here, in a nice new OIDC multi-tenancy concept doc you introduce with this PR, I believe we should highlight the multi-tenancy itself - which is done very well.
What do you think ?

Reply:
We could reference other guides if appropriate. 

I would suggest We spin this guide as a story through multi-tenancy, as sen from the developer's perspective. 

Will take a look and nudge descriptions in other guides so they harmomize

Reply:
@lennartj What can make it simpler is for this concept doc to focus purely on explaining the multitenancy idea; a lot what you typed in that regard would make it a perfect concept doc wh8ch would accompany the already existing multitenancy how to doc.
Everything related to the actual token treatment, should be applied where necessary in follow  up PRs to the bearer and code flow guides IMHO, it will be easier to merge this PR that way 🙂